src/software/ai/move_rule/move_rule_manager.cpp

```diff
@@ -0,0 +1,103 @@
+#include "software/ai/move_rule/move_rule_manager.h"
+
+#include "software/ai/hl/stp/tactic/all_tactics.h"
+
+std::set<MoveRule> MoveRuleManager::getMoveRules(const GameState &game_state,
+                                                 const Tactic &tactic)
+{
+    std::set<MoveRule> current_move_rules;
+
+    // function for adding move rules
+    std::function<void(MoveRule)> add_move_rule = [&](MoveRule rule) {
+        current_move_rules.erase(rule);
+    };
+    addCurrentMoveRulesFromGameState(game_state, add_move_rule);
```

Comment:
Why don't we just make `rules_to_whitelist` a class member that the visit function can modify? Then we can write:
```
    std::set<MoveRule> move_rules = getCurrentMoveRulesFromGameState(game_state);
    tactic.accept(*this);
    for (const auto& rule : rules_to_whitelist){
        move_rules.erase(rule);
    }
    return move_rules;

Reply:
Yeah I guess that's the more idiomatic way for c++